transaction/builder/src/transaction_builder.rs

```diff
@@ -2801,36 +2718,82 @@ pub mod transaction_builder_tests {
                     )
                     .unwrap();
 
-                assert!(
-                    transaction_builder.set_fee(Mob::MINIMUM_FEE * 4).is_err(),
-                    "setting fee after change output should be rejected"
-                );
+                transaction_builder
+                    .add_output(
+                        Amount::new(Mob::MINIMUM_FEE, token_id),
+                        &recipient_address,
+                        &mut rng,
+                    )
+                    .unwrap();
+                let result =
+                    transaction_builder.build(&NoKeysRingSigner {}, memo_builder, &mut rng);
 
-                assert!(
-                    transaction_builder
-                        .add_output(
-                            Amount::new(Mob::MINIMUM_FEE, token_id),
-                            &recipient_address,
-                            &mut rng
-                        )
-                        .is_err(),
-                    "Adding another output after change output should be rejected"
+                assert_matches!(
+                    result,
+                    Err(TxBuilderError::NewTx(NewTxError::Memo(
+                        NewMemoError::OutputsAfterChange
+                    )))
                 );
+            }
 
-                assert!(
-                    transaction_builder
-                        .add_change_output(
-                            Amount::new(change_value, token_id),
-                            &sender_change_dest,
-                            &mut rng
-                        )
-                        .is_err(),
-                    "Adding a second change output should be rejected"
+            // Test that adding a  change output after the change output causes an error as
+            // expected
```

Comment:
I think the "x after y" terminology doesn't make as much sense with the memo being applied at build time, instead of per `add_output()` call

```suggestion
            // Multiple change outputs cause error
```